tools/clang/test/SemaHLSL/rayquery-omm-type-diag.hlsl

```diff
@@ -0,0 +1,12 @@
+// RUN: %dxc -T vs_6_9 -E RayQueryTests -verify %s
+// RUN: %dxc -T vs_6_5 -E RayQueryTests2 -verify %s
+
+// validate 2nd template argument flags
+// expected-error@+1{{When using 'RAY_FLAG_FORCE_OMM_2_STATE' in RayFlags, RayQueryFlags must have RAYQUERY_FLAG_ALLOW_OPACITY_MICROMAPS set.}}
+typedef RayQuery<RAY_FLAG_FORCE_OMM_2_STATE> BadRayQuery;
+// expected-error@+1{{When using 'RAY_FLAG_FORCE_OMM_2_STATE' in RayFlags, RayQueryFlags must have RAYQUERY_FLAG_ALLOW_OPACITY_MICROMAPS set.}}
+typedef RayQuery<RAY_FLAG_FORCE_OMM_2_STATE, 0> BadRayQuery2;
+
+BadRayQuery rayQuery0a;
```

Comment:
Oh, we should probably have prevented declaring RayQuery variables as constant buffer values, that's a bug that there's no diagnostic here.  Maybe for this change you can add "static" and we can file an issue for that bug?

Reply:
Created an issue here:
https://github.com/microsoft/DirectXShaderCompiler/issues/7202

Reply:
Filed issue: #7203.